src/InputWithLabel.js

```diff
@@ -0,0 +1,23 @@
+import React, { useEffect, useRef } from "react";
+
+const InputWithLabel = ({ todoValue, onInputChange, children }) => {
+  const inputRef = useRef();
+  useEffect(() => {
+    inputRef.current.focus();
+  }, [todoValue]);
```

Comment:
Nitpick:
This dependency list adds a manual control for when the input is focused (every time the `todoValue` has changed). In this case, this isn't necessary. Instead, this useEffect can go without a dependency list and achieve the same result:

> [Add Auto-Focus to Input (lesson 1.6)](https://github.com/Code-the-Dream-School/react/wiki/Lesson-1.6#add-auto-focus-to-input)
> - [ ] Define a useEffect React hook without dependency list

The one advantage to your code is that if another piece of state were to re-render this component, then it wouldn't automatically re-focus the input. That said, since this a requirement for the assignment, I'd recommend updating it to not have a dependency list.

Reply:
Went back and cleaned it up, thank you for your help!!